lib/MetaCPAN/Script/Release.pm

```diff
@@ -490,6 +498,73 @@ sub import_tarball {
     }
 }
 
+sub find_main_module {
+    my $self = shift;
+    my ( $mod, $distribution ) = @_;
+
+    my @modules = @{$mod};
+
+    my $min_level          = 100;
+    my @distribution_parts = split /-/, $distribution;
+    my $distribution_name  = join '', @distribution_parts;
+
+    if ( scalar @modules eq 1 ) {
+
+        # there is only one module and it will become the main_module
+        my $main_module = shift @modules;
+        $self->main_module( $main_module->module );
+        return;
+    }
+    else {
+        foreach my $file ( sort @modules ) {
+            my @module_parts = split /::/, $file->module->[0]->{name};
+            my $module_name = join '', @module_parts;
+
+            # the module has the exact name as the ditribution
+            if ( $module_name eq $distribution_name ) {
+                $self->main_module( $file->module );
+                return;
+            }
+
+            # the module contains the ditribution name
+            if ( $module_name =~ /$distribution_name/ ) {
+                $self->main_module( $file->module );
+                return;
+            }
+
+            # the distribution has modules on different levels
+            # the main_module is the one with minimum level
+            if ( $file->level lt $min_level ) {
```

Comment:
I'm not going to look at this logic too closely right now.  I think after you add some tests it will become clearer if this is the right approach.